x/dualstaking/keeper/grpc_query_delegator_providers.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/x/dualstaking/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) DelegatorProviders(goCtx context.Context, req *types.QueryDelegatorProvidersRequest) (res *types.QueryDelegatorProvidersResponse, err error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	epoch := uint64(ctx.BlockHeight())
+	if req.WithPending {
+		epoch, err = k.getNextEpoch(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	providers, err := k.GetDelegatorProviders(ctx, req.Delegator, epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	var delegations []types.Delegation
+	for _, provider := range providers {
+		indices := k.delegationFS.GetAllEntryIndicesWithPrefix(ctx, provider)
```

Comment:
won't work, because the key is currently <provider, chain-id, delegator>. for this to work, the key needs to be <provider, delegator, chain-id>.

we can change the key order now; but keep in mind that feature will change to be per provider (instead of per provider per chain-id). when that happens, it would work (as I mentioned in the other comment).

Reply:
do we lose anything from changing it now?

Reply:
done. see https://github.com/lavanet/lava/commit/9ac02e1108b5717f64fbd634f72bb539a88c4bdf